src/main/webapp/app/core/auth/account.service.ts

```diff
@@ -169,7 +171,25 @@ export class AccountService implements IAccountService {
      * @param course
      */
     isAtLeastTutorInCourse(course?: Course): boolean {
-        return this.hasGroup(course?.instructorGroupName) || this.hasGroup(course?.teachingAssistantGroupName) || this.hasAnyAuthorityDirect([Authority.ADMIN]);
+        return (
+            this.hasGroup(course?.instructorGroupName) ||
+            this.hasGroup(course?.editorGroupName) ||
+            this.hasGroup(course?.teachingAssistantGroupName) ||
+            this.hasAnyAuthorityDirect([Authority.ADMIN])
+        );
+    }
+
+    /**
+     * checks if the currently logged in user is at least editor in the given course
+     * @param course
+     */
+    isAtLeastEditorInCourse(course?: Course): boolean {
+        return (
+            this.hasGroup(course?.instructorGroupName) ||
+            this.hasGroup(course?.editorGroupName) ||
+            this.hasGroup(course?.teachingAssistantGroupName) ||
```

Comment:
Shouldn't you remove this line since an editor has more rights than a teaching assistant?

Reply:
Thank you, we've been searching this bug for at least one day...